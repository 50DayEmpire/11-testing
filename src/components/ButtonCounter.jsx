const ButtonCounter = ({ value, action }) => {
  switch (value) {
    case 1:
      return <button onClick={action}>+1</button>;
    case 0:
      return <button onClick={action}>Reset</button>;
    case -1:
      return (
        <button aria-label="disminuir" onClick={action}>
          -1
        </button>
      );
    default:
      break;
  }
};

export default ButtonCounter;
