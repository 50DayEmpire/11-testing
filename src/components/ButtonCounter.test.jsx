import { beforeEach } from "vitest";
import ButtonCounter from "./ButtonCounter";
import { render, screen } from "@testing-library/react";

describe("<ButtonCounter />", () => {
  const action = vi.fn();
  beforeEach(() => {
    render(<ButtonCounter value={1} action={action} />);
  });

  it("Snapshot de todas las pruebas", () => {
    const button = screen.getByRole("button");
    expect(button).toMatchSnapshot();
  });

  it("cree un boton correctamente", () => {
    const button = screen.getByRole("button");
    expect(button.textContent).toBe("+1");
  });
});
