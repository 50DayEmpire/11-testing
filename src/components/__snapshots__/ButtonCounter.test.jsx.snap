// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ButtonCounter /> > Snapshot de todas las pruebas 1`] = `
<button>
  +1
</button>
`;
